### PR TITLE
Codechange: Switch to C++17 on all platforms.

### PR DESCRIPTION
**Pull request by frosch123**

### CODE REVIEW

**Comment by LordAro** [comment]
Documentation needs updating too ;)

**Comment by TrueBrain** [comment]
I was thinking that it might be wise to keep this for after 1.11 is branched, just so there is more time to make use of it etc :)

**Comment by frosch123** [comment]
According to `https://github.com/pybind/` snip `python_example/issues/` snip `44#issuecomment` snip `-714842598` (found by @michicc ) all interesting parts of C++17 are broken on OSX <= 10.14.

The compile farm seems to use 10.15.

I have no idea why this should affect OSX users who do not compile themself.

File: CMakeLists.txt [hunk 1]
```diff
@@ -32,13 +32,8 @@ set_directory_options()
...
 set_static_if_needed()
 
-if(MSVC)
-    # C++17 for MSVC
-    set(CMAKE_CXX_STANDARD 17)
-else()
-    # C++11 for all other targets
-    set(CMAKE_CXX_STANDARD 11)
-endif()
+# C++17 for all targets
```
Comment (TrueBrain) [inline CMakeLists.txt]: Pretty sure we an do without that comment now :D

**Review by TrueBrain — APPROVED** [review]
I have been trying to find a reason to deny this patch, but I cannot find one :(

**Pull request merged**